builder/amazon/common/step_get_password.go

```diff
@@ -91,16 +90,13 @@ WaitLoop:
 			"Password (since debug is enabled): %s", s.Comm.WinRMPassword))
 	}
 	// store so that we can access this later during provisioning
-
-	commonhelper.SetSharedState("winrm_password", s.Comm.WinRMPassword, s.BuildName)
+	state.Put("winrm_password", s.Comm.WinRMPassword)
```

Comment:
so `commonhelper.SetSharedState` will store something in some sort of plugin (component) shared file and `state.Put` stores things in a map ? I really like the idea of removing `commonhelper.SetSharedState``

Reply:
Yeah, I really want to remove the SharedState file hack. I have removed it for everything except the internal port and IP handling; that can probably go too but doesn't need to be part of this PR.